data_diff/table_segment.py

```diff
@@ -15,6 +17,71 @@
 RECOMMENDED_CHECKSUM_DURATION = 20
 
 
+def split_key_space(min_key: DbKey, max_key: DbKey, count: int) -> List[DbKey]:
+    assert min_key < max_key
+
+    if max_key - min_key <= count:
+        count = 1
```

Comment:
Yes, this is just taking care of the edge case when the distance is smaller than the count. Then it doesn't make sense to split it further. hence count =1.

The point was just to prevent errors further down the line.